chapters/VK_EXT_present_timing/queries.adoc

```diff
@@ -0,0 +1,481 @@
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to a structure extending this
+    structure.
+  * pname:presentId is an application-provided value that was given to a
+    previous fname:vkQueuePresentKHR command via
+    slink:VkPresentIdKHR::pname:pPresentIds.
+    It can: be used to uniquely identify a previous present with the
+    flink:vkQueuePresentKHR command. If no slink:VkPresentIdKHR was provided
+    or if the specified present id value was zero, pname:presentId is zero.
+  * pname:targetPresentTime is an application-provided value that was given
+    to a previous flink:vkQueuePresentKHR command via
+    slink:VkPresentTimeEXT::pname:targetPresentTime.
+    If non-zero, it was used by the application to indicate that an image
+    not be presented any sooner than pname:targetPresentTime, relative to
+    the stage specified in
+    slink:VkSwapchainPresentTimingCreateInfoEXT::pname:presentStageTarget.
+    If the pname:swapchain is using relative present times, it is zero and
+    the actual present duration can: be obtained by comparing the
+    pname:pPresentStages results of consecutive presents.
+  * pname:presentStageCount is a count of items contained in
+    pname:pPresentStages.
+  * pname:pPresentStages is `NULL` or a pointer to an array of
+    slink:VkPresentStageTimeEXT providing timing information for each of the
+    present stages requested in
+    slink:VkSwapchainPresentTimingCreateInfoEXT::pname:presentStageQueries.
+  * pname:timingPropertiesChanged is ename:VK_TRUE if the swapchain's
+    timing properties have changed since the last time those
+    properties were queried with
+    flink:vkGetSwapchainTimingPropertiesEXT, otherwise the properties
+    have not changed.
+  * pname:timeDomainChanged is ename:VK_TRUE if the time domain enabled for
+    the swapchain is not currently available.
+  * pname:reportComplete is ename:VK_TRUE if the presentation engine
+    has reported all the requested results in pname:pPresentStages.
+
+Timing information for some present stages may: have a time value of 0,
+indicating that results for that present stage are not available.
+
+For systems with multiple entities operating downstream of the presentation
+engine, such as multiple displays, pname:pPresentStages will return timing
+results for at least one entity which has been affected by the presentation.
+
+If pname:pPresentStages is `NULL`, then the number of available present
+stages reports is returned in pname:presentStageCount. Otherwise,
+pname:presentStageCount must: be set by the application to the number of
+elements in the pname:pPresentStages array, and on return the variable is
+overwritten with the number of entries actually written to
+pname:pPresentStages. If the value of pname:presentStageCount is less than
+the number of available present stages reports, at most
+pname:presentStageCount will be written to pname:pPresentStages.
```

Comment:
If presentStageCount is less than the available stages of feedback, is VK_INCOMPLETE returned? If so, how does the application differentiate between this incomplete, and a "I'm not done with the other stages yet" incomplete? It seems like both could happen at once. Is there an interaction between reportComplete, the return value (VK_SUCCESS, VK_INCOMPLETE) for the query associated with this struct instance, and whether the results fit in presentStageCount?

Perhaps it should be required that the application set presentStageCount equal to the number of stages it requested at swapchain creation time?

Reply:
> Perhaps it should be required that the application set presentStageCount equal to the number of stages it requested at swapchain creation time?

That is probably best, yes.